processing/data_collection/gazette/spiders/sc_joinville.py

```diff
@@ -0,0 +1,60 @@
+import re
+import dateparser
+
+from datetime import datetime
+from scrapy import Request, Spider
+from gazette.items import Gazette
+
+class GazetteElement:
+    DATE_CSS = 'span.article-date::text'
+    MUNICIPALITY_ID = '4209102'
+    PDF_URL = 'https://www.joinville.sc.gov.br{}'
+
+    def __init__(self, element):
+        self.element = element
+
+    def date(self):
+        date_str = self.element.css(self.DATE_CSS)[-1].extract()
+        return dateparser.parse(date_str, languages=['pt']).date()
+
+    def url(self):
+        path = self.element.css('a::attr(href)').extract_first()
+        return self.PDF_URL.format(path)
+
+    def to_gazette(self):
+        return Gazette(
+            date=self.date(),
+            file_urls=[self.url()],
+            is_extra_edition=False,
+            municipality_id=self.MUNICIPALITY_ID,
+            power='executive_legislature',
+            scraped_at=datetime.utcnow(),
+        )
+
+class ScJoinvilleSpider(Spider):
+    ELEMENT_CSS = 'ul.jornal li'
+    NEXT_PAGE_CSS = 'ul.pagination li.next a::attr(href)'
+
+    allowed_domains = ['joinville.sc.gov.br']
+    name = 'sc_joinville'
+    start_urls = ['https://www.joinville.sc.gov.br/jornal/index/page/1']
+
+    def parse(self, response):
+        """
+        @url http://www.joinville.sc.gov.br/jornal/index/page/1
+        @returns requests 1
+        @scrapes date file_urls is_extra_edition municipality_id power scraped_at
+        """
+
+        for element in self.get_gazettes_elements(response):
+            yield GazetteElement(element).to_gazette()
```

Comment:
Thats cool! Didn't know the ItemLoader. I extracted the code to a class instead of putting everything in the parser method to improve readability and comprehension of others that will read this code in the future. I liked the ItemLoader suggestion, but didn't understand very well how we could use it properly here. This is something we could use in all other scrapers. Let's talk with others about this.

Reply:
Everything in `parse` method is readable enough. Anyone with some experience with Scrapy will expect to find the things you included in this object in the `parse` method. I believe this is including an unnecessary complexity and more difficult to understand the purpose of the object (exactly the opposite of your goal).